### PR TITLE
Generalize Option Class

### PR DESCRIPTION
**Pull request by wbinventor**
This short PR generalizes the `openmoc.Options` class to make it sub-classable. This can be nice for users who want to add their own options for certain types of simulations, but leverage the structure already implemented in `openmoc.Options`.

### CODE REVIEW

File: openmoc/options.py [hunk 1]
```diff
@@ -65,30 +75,69 @@ def __init__(self):
...
+    @property
+    def num_omp_threads(self):
+        return self._num_omp_threads
+
+    @property
+    def num_thread_blocks(self):
+        return self._num_thread_blocks
+
+    @property
+    def num_threads_per_block(self):
```
Comment (samuelshaner) [inline openmoc/options.py]: All other attributes besides `_num_gpu_threads` use their name in the property method getter. What do you think about having this `def num_gpu_threads(self):`?

Comment (wbinventor) [inline openmoc/options.py]: Oh yes, I agree and will fix it accordingly now.

File: openmoc/options.py [hunk 2]
```diff
@@ -58,37 +58,86 @@ def __init__(self):
...
 
+        self._short_args = 'hfa:s:i:c:t:b:g:r:l:'
+        self._long_args = ['help',
+                           'num-azim=',
+                           'track-spacing=',
+                           'tolerance=',
+                           'max-iters=',
+                           'num-omp-threads=',
+                           'num-thread-blocks=',
+                           'num-gpu-threads=']
```
Comment (samuelshaner) [inline openmoc/options.py]: can it be changed here as well?

File: openmoc/options.py [hunk 3]
```diff
@@ -146,32 +196,4 @@ def parseArguments(self):
             elif opt in ('-b', '--num-thread-blocks'):
                 self._num_thread_blocks = int(arg)
             elif opt in ('-g', '--num-gpu-threads'):
```
Comment (samuelshaner) [inline openmoc/options.py]: and here

File: openmoc/options.py [hunk 4]
```diff
@@ -122,14 +171,15 @@ def parseArguments(self):
...
-                num_gpu_threadblocks += 'The number of GPU threadblocks\n'
-                print(num_gpu_threadblocks)
+                num_threadblocks += 'The number of GPU threadblocks\n'
+                print(num_threadblocks)
 
-                num_gpu_threads = '\t{: <35}'.format('-g, --num-gpu-threads=<64>')
-                num_gpu_threads += 'The number of GPU threads per block\n'
-                print(num_gpu_threads)
+                num_threads_per_block = \
+                    '\t{: <35}'.format('-g, --num-gpu-threads=<64>')
```
Comment (samuelshaner) [inline openmoc/options.py]: and here

File: openmoc/options.py (hunk 2 above)
Comment (wbinventor) [inline openmoc/options.py]: agreed, it's standardized here and below now.

**Comment by samuelshaner** [comment]
Looks good. Thanks for another short and easy to review PR @wbinventor!

**Pull request merged**